packages/puppeteer-core/src/cdp/WebWorker.ts

```diff
@@ -79,4 +82,21 @@ export class CdpWebWorker extends WebWorker {
   get client(): CDPSession {
     return this.#client;
   }
+
+  override async close(): Promise<void> {
+    try {
+      // For service_workers. TODO: perhaps we need to check the worker type
+      // directly.
+      await this.client.connection()?.send('Target.closeTarget', {
+        targetId: this.#id,
+      });
+      await this.client.connection()?.send('Target.detachFromTarget', {
+        sessionId: this.client.id(),
+      });
+    } catch {
+      await this.evaluate(() => {
```

Comment:
This will error for service workers. We should just close the target.

Reply:
Target.closeTarget is not supported for all worker types

Reply:
Then can we try `this.evaluate` before doing target.close?